components/eventing-controller/pkg/commander/nats/nats.go

```diff
@@ -0,0 +1,145 @@
+		return err
+	}
+	if err := eventingv1alpha1.AddToScheme(scheme); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Commander implements the Commander interface.
+type Commander struct {
+	cancel          context.CancelFunc
+	envCfg          env.NatsConfig
+	restCfg         *rest.Config
+	enableDebugLogs bool
+	metricsAddr     string
+	mgr             manager.Manager
+	backend         handlers.MessagingBackend
+}
+
+// NewCommander creates the Commander for BEB and initializes it as far as it
+// does not depend on non-common options.
+func NewCommander(restCfg *rest.Config, enableDebugLogs bool, metricsAddr string, maxReconnects int, reconnectWait time.Duration) *Commander {
+	return &Commander{
+		envCfg:          env.GetNatsConfig(maxReconnects, reconnectWait), // TODO Harmonization.
+		restCfg:         restCfg,
+		enableDebugLogs: enableDebugLogs,
+		metricsAddr:     metricsAddr,
+	}
+}
+
+// Init implements the Commander interface.
+func (c *Commander) Init(mgr manager.Manager) error {
+	if len(c.envCfg.Url) == 0 {
+		return fmt.Errorf("env var URL must be a non-empty value")
+	}
+	c.mgr = mgr
+	return nil
+}
+
+// Start implements the Commander interface and starts the commander.
+func (c *Commander) Start() error {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	c.cancel = cancel
+	dynamicClient := dynamic.NewForConfigOrDie(c.restCfg)
+	applicationLister := application.NewLister(ctx, dynamicClient)
+	natsReconciler := subscription.NewReconciler(
+		ctx,
+		c.mgr.GetClient(),
+		applicationLister,
+		c.mgr.GetCache(),
+		ctrl.Log.WithName("reconciler").WithName("Subscription"),
+		c.mgr.GetEventRecorderFor("eventing-controller-nats"),
+		c.envCfg,
+	)
+	c.backend = natsReconciler.Backend
+	if err := natsReconciler.SetupUnmanaged(c.mgr); err != nil {
+		return fmt.Errorf("unable to setup the NATS subscription controller: %v", err)
+	}
+	return nil
+}
+
+// Stop implements the Commander interface and stops the commander.
+func (c *Commander) Stop() error {
+	c.cancel()
+
+	return c.cleanup()
+}
+
+// clean removes all NATS artifacts.
+func (c *Commander) cleanup() error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	logger := ctrl.Log.WithName("eventing-controller-nats-cleaner").WithName("Subscription")
+	var natsBackend *handlers.Nats
+	var ok bool
+	var natsBackendErr error
+	if natsBackend, ok = c.backend.(*handlers.Nats); !ok {
+		logger.Error(natsBackendErr, "no NATS backend exists")
+	}
+	// Fetch all subscriptions.
+	dynamicClient := dynamic.NewForConfigOrDie(c.restCfg)
+	subscriptionsUnstructured, err := dynamicClient.Resource(handlers.GroupVersionResource()).Namespace(corev1.NamespaceAll).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	subs, err := handlers.ToSubscriptionList(subscriptionsUnstructured)
+	if err != nil {
+		return err
+	}
+
+	statusDeletionResult := make(map[string]error)
+	subDeletionResult := make(map[string]error)
```

Comment:
Why?

Reply:
Refactored with in place logging and removed the error maps.